### PR TITLE
Preparation for Issue due dates (go-gitea/gitea#2533)

### PR DESCRIPTION
**Pull request by kolaente**
I'm currently implementing go-gitea/gitea#2533, this pr should be merged before go-gitea/gitea#3794.

Blocks go-gitea/gitea#3794.

### CODE REVIEW

**Comment by lunny** [comment]
LGTM

File: gitea/issue.go [hunk 1]
```diff
@@ -114,12 +118,13 @@ func (c *Client) CreateIssue(owner, repo string, opt CreateIssueOption) (*Issue,
...
-	Assignees []string `json:"assignees"`
-	Milestone *int64   `json:"milestone"`
-	State     *string  `json:"state"`
+	Title     string     `json:"title"`
+	Body      *string    `json:"body"`
+	Assignee  *string    `json:"assignee"`
+	Assignees []string   `json:"assignees"`
+	Milestone *int64     `json:"milestone"`
+	State     *string    `json:"state"`
+	Deadline  *time.Time `json:"due_date"`
```
Comment (jonasfranz) [inline gitea/issue.go]: `deadline` should also be added to CreateIssueOption

File: gitea/issue.go [hunk 2]
```diff
@@ -94,6 +96,8 @@ type CreateIssueOption struct {
 	// username of assignee
 	Assignee  string   `json:"assignee"`
 	Assignees []string `json:"assignees"`
+	// deadline
+	Deadline *time.Time `json:"due_date"`
```
Comment (jonasfranz) [inline gitea/issue.go]: Add `// swagger:strfmt date-time`

File: gitea/issue.go (hunk 1 above)
Comment (jonasfranz) [reply gitea/issue.go]: Add `// swagger:strfmt date-time`

File: gitea/pull.go [hunk 3]
```diff
@@ -82,14 +85,15 @@ func (c *Client) GetPullRequest(owner, repo string, index int64) (*PullRequest,
...
-	Labels    []int64  `json:"labels"`
+	Head      string     `json:"head" binding:"Required"`
+	Base      string     `json:"base" binding:"Required"`
+	Title     string     `json:"title" binding:"Required"`
+	Body      string     `json:"body"`
+	Assignee  string     `json:"assignee"`
+	Assignees []string   `json:"assignees"`
+	Milestone int64      `json:"milestone"`
+	Labels    []int64    `json:"labels"`
+	Deadline  *time.Time `json:"due_date"`
```
Comment (jonasfranz) [inline gitea/pull.go]: Add `// swagger:strfmt date-time`

File: gitea/pull.go [hunk 4]
```diff
@@ -105,13 +109,14 @@ func (c *Client) CreatePullRequest(owner, repo string, opt CreatePullRequestOpti
...
-	Labels    []int64  `json:"labels"`
-	State     *string  `json:"state"`
+	Title     string     `json:"title"`
+	Body      string     `json:"body"`
+	Assignee  string     `json:"assignee"`
+	Assignees []string   `json:"assignees"`
+	Milestone int64      `json:"milestone"`
+	Labels    []int64    `json:"labels"`
+	State     *string    `json:"state"`
+	Deadline  *time.Time `json:"due_date"`
```
Comment (jonasfranz) [inline gitea/pull.go]: Add `// swagger:strfmt date-time`

**Review by jonasfranz — CHANGES_REQUESTED** [review]

File: gitea/issue.go (hunk 1 above)
Comment (kolaente) [reply gitea/issue.go]: > deadline should also be added to CreateIssueOption

It is, see line 99.

**Comment by kolaente** [comment]
@JonasFranzDEV fixed.

**Comment by jonasfranz** [comment]
LGTM

**Pull request merged**